packages/core/src/framebuffer/FramebufferSystem.ts

```diff
@@ -98,8 +98,32 @@ export class FramebufferSystem implements ISystem
         else
         {
             // WebGL2
-            // cache possible MSAA samples
-            this.msaaSamples = gl.getInternalformatParameter(gl.RENDERBUFFER, gl.RGBA8, gl.SAMPLES);
+            this.msaaSamples = {};
+
+            // Don't use more than `MAX_SAMPLES`: some devices do not allow to create renderbuffers
+            // with more than `MAX_SAMPLES` despite the fact that it is a valid number of samples
+            // according to `getInternalformatParameter`.
+            const maxSamples = Math.min(gl.getParameter(gl.MAX_SAMPLES) as number, MSAA_QUALITY.HIGH);
```

Comment:
shall we move this to a separate function? FrameBufferSystem is getting pretty beefy!